examples/catalog/lib/basic_app_bar.dart

```diff
@@ -26,15 +27,18 @@ class _BasicAppBarSampleState extends State<BasicAppBarSample> {
         appBar: AppBar(
           title: const Text('Basic AppBar'),
           actions: <Widget>[
-            IconButton( // action button
+            IconButton(
+              // action button
               icon: Icon(choices[0].icon),
               onPressed: () { _select(choices[0]); },
             ),
-            IconButton( // action button
+            IconButton(
+              // action button
               icon: Icon(choices[1].icon),
               onPressed: () { _select(choices[1]); },
             ),
-            PopupMenuButton<Choice>( // overflow menu
+            PopupMenuButton<Choice>(
+              // overflow menu
               onSelected: _select,
```

Comment:
i think these three changes make this section less readable. It makes the comments look like they're talking about the `icon`, `icon`, and `onSelected` arguments, instead of being about the `IconButton`, `IconButton`, and `PopupMenuButton<Choice>` widgets, which is what they're supposed to be about.

Reply:
in general i would recommend against changing what construct a comment is attached to in this way.

Reply:
when i made my attempt at a formatter, i treated comments specially -- they were annotations attached to the "active" thing (in this case, the constructor calls), and when i reserialised, they'd get attached to the output done for that construct. One could imagine distinguishing the case where the comment is on a line on its own (gets associated with the next line's construct), vs being at the end of a line (gets associated with the thing before it).

Reply:
I agree. I will try to have a look.

Reply:
Filed https://github.com/dart-lang/dart_style/issues/798